tests/Deputy/Subscriber/BindAdherentDistrictSubscriberTest.php

```diff
@@ -15,44 +15,52 @@
 use PHPUnit\Framework\TestCase;
 use Ramsey\Uuid\Uuid;
 
-class ChangeAddressSubscriberTest extends TestCase
+class BindAdherentDistrictSubscriberTest extends TestCase
 {
     private $manager;
 
     /* @var DistrictRepository */
     private $districtRepository;
 
-    /* @var ChangeAddressSubscriber */
+    /* @var BindAdherentDistrictSubscriber */
     private $subscriber;
 
-    public function testOnAdherentAccountRegistrationCompletedSucceeds()
+    /**
+     * @dataProvider provideReferentTagHasCount
+     */
+    public function testOnAdherentAccountRegistrationCompletedSucceeds(array $referentTags, int $count)
     {
         $adherent = $this->createAdherent();
-        $referentTag = $this->createReferentTag();
 
         $this->assertInstanceOf(ReferentTaggableEntity::class, $adherent);
         $this->assertSame(0, $adherent->getReferentTags()->count());
 
-        $this->manager->expects($this->once())->method('flush');
-        $this->districtRepository->expects($this->once())->method('findDistrictReferentTagByCoordinates')->willReturn($referentTag);
+        if ($count > 0) {
+            $this->manager->expects($this->once())->method('flush');
+        }
+        $this->districtRepository->expects($this->once())->method('findDistrictReferentTagByCoordinates')->willReturn($referentTags);
         $this->subscriber->updateReferentTagWithDistrict(new AdherentAccountWasCreatedEvent($adherent));
 
-        $this->assertSame(1, $adherent->getReferentTags()->count());
+        $this->assertSame($count, $adherent->getReferentTags()->count());
     }
 
-    public function testOnAdherentProfileUpdatedSuccessfully()
+    /**
+     * @dataProvider provideReferentTagHasCount
+     */
+    public function testOnAdherentProfileUpdatedSuccessfully(array $referentTags, int $count)
     {
         $adherent = $this->createAdherent();
-        $referentTag = $this->createReferentTag();
 
         $this->assertInstanceOf(ReferentTaggableEntity::class, $adherent);
         $this->assertSame(0, $adherent->getReferentTags()->count());
 
-        $this->manager->expects($this->once())->method('flush');
-        $this->districtRepository->expects($this->once())->method('findDistrictReferentTagByCoordinates')->willReturn($referentTag);
+        if ($count > 0) {
```

Comment:
I think we can do:
```
if (0 < $count =  count($referentTags)) {
  // ...
}
```
and remove `int $count` argument from method signature, WDYT?